node/silkworm/db/mdbx.cpp

```diff
@@ -20,22 +20,25 @@ namespace silkworm::db {
 
 namespace detail {
 
-    //! \brief Returns data of current cursor position or moves it to begin or end of table on behalf of provided
-    //! direction.
-    //! \param [in] _c : A reference to an open cursor
-    //! \param [in] _d : Direction cursor should have
-    //! \return ::mdbx::cursor::move_result
-    static inline ::mdbx::cursor::move_result cursor_data_factory(::mdbx::cursor& _c, CursorMoveDirection _d) {
-        if (_c.eof()) {
-            return (_d == CursorMoveDirection::Forward) ? _c.to_first(false) : _c.to_last(false);
+    //! \brief Returns data of current cursor position or moves it to the beginning or the end of the table based on
+    //! provided direction if the cursor is not positioned.
+    //! \param [in] c : A reference to an open cursor
+    //! \param [in] d : Direction cursor should have \return ::mdbx::cursor::move_result
+    static inline ::mdbx::cursor::move_result data_of_positioned_cursor(::mdbx::cursor& c, CursorMoveDirection d) {
```

Comment:
This is semantically wrong.
`data_of_positioned_cursor == cursor.current`

Here instead the function

1. cursor is moved
2. data is pulled

Reply:
Fair enough, `adjust_cursor_postion_if_unpositioned_and_return_data` is more accurate

Reply:
Ok. Maybe 8f88f5b ?
We can start a new coding pattern ?

Reply:
Well, function name should reflect its intention. "erase" clearly means that it erases something. If I want more detail, then I can read the doc or the source code. But I get some rough idea from "erase" alone. The problem with "cursor_data_factory" is that doesn't give a hint (at least not to me) about the function's intention to move the cursor to the beginning/end of the table if the cursor is not positioned.

Reply:
Well. I'll stop fooling around here cause we're building a case over the name of a function which :

1. Is extensively commented
2. Is in detail namespace of mdbx.cpp file (so not exposed for other source files)
3. Used only in the same file where paging up few lines is not a big effort
4. has "factory" in its name which clearly states a **creational pattern**

I wish we could not waste more hours on things like this cause, honestly, our code is disseminated of things like this (eg. hash() is the verb implying an action - to hash something - or the getter of a property ?). 
Anyway ... your will be done and let's move on.